pkg/authorization/registry/clusterrole/proxy.go

```diff
@@ -0,0 +1,165 @@
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
+	"github.com/openshift/origin/pkg/authorization/registry/util"
+)
+
+type REST struct {
+	client   restclient.Interface
+	resource schema.GroupResource
+}
+
+func NewREST(client restclient.Interface) *REST {
+	return &REST{client: client, resource: authorizationapi.Resource("clusterrole")}
+}
+
+func (s *REST) New() runtime.Object {
+	return &authorizationapi.ClusterRole{}
+}
+func (s *REST) NewList() runtime.Object {
+	return &authorizationapi.ClusterRoleList{}
+}
+
+func (s *REST) List(ctx apirequest.Context, options *metainternal.ListOptions) (runtime.Object, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	optv1 := metav1.ListOptions{}
+	if err := metainternal.Convert_internalversion_ListOptions_To_v1_ListOptions(options, &optv1, nil); err != nil {
+		return nil, err
+	}
+
+	roles, err := client.List(optv1)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &authorizationapi.ClusterRoleList{}
+	for _, curr := range roles.Items {
+		role, err := util.ClusterRoleFromRBAC(&curr)
+		if err != nil {
+			return nil, err
+		}
+		ret.Items = append(ret.Items, *role)
+	}
+	return ret, nil
+}
+
+func (s *REST) Get(ctx apirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ret, err := client.Get(name, *options)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, apierrors.NewNotFound(s.resource, name)
+		}
+		return nil, err
+	}
+
+	role, err := util.ClusterRoleFromRBAC(ret)
+	if err != nil {
+		return nil, err
+	}
+	return role, nil
+}
+
+func (s *REST) Delete(ctx apirequest.Context, name string, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if err := client.Delete(name, options); err != nil {
+		return nil, false, err
+	}
+
+	return &metav1.Status{Status: metav1.StatusSuccess}, true, nil
+}
+
+func (s *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runtime.Object, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	convertedObj, err := util.ClusterRoleToRBAC(obj.(*authorizationapi.ClusterRole))
+	if err != nil {
+		return nil, err
+	}
+
+	ret, err := client.Create(convertedObj)
+	if err != nil {
+		return nil, err
+	}
+
+	role, err := util.ClusterRoleFromRBAC(ret)
+	if err != nil {
+		return nil, err
+	}
+	return role, nil
+}
+
+func (s *REST) Update(ctx apirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	old, err := client.Get(name, metav1.GetOptions{})
```

Comment:
if initializers use `Update()` to initialize objects (which I think they do), we may need to pass `GetOptions{IncludeUninitialized: true}` here... @smarterclayton, does that sound right?